core/server/services/users.js

```diff
@@ -17,17 +17,59 @@ const path = require('path');
  * @prop {Object} ApiKey.NotFoundError
  * @prop {Object} User
  * @prop {(Object) => Promise} User.destroy
+ * @prop {(Object) => Promise} User.findAll
+ * @prop {Object} Session
+ * @prop {(Object) => Promise} Session.findAll
+ */
+
+/**
+ * @typedef {Object} IAuth
+ * @prop {Object} setup
+ * @prop {(isComplete: boolean) => () => Promise} setup.assertSetupCompleted
+ * @prop {Object} passwordreset
+ * @prop {(email: string, apiSettings: Object, transcation?: Object) => Promise<string>} passwordreset.generateToken
+ * @prop {(token: string, apiMail: Object) => Promise} passwordreset.sendResetNotification
  */
 
 class Users {
     /**
      * @param {Object} dependencies
      * @param {IdbBackup} dependencies.dbBackup
      * @param {IModels} dependencies.models
+     * @param {IAuth} dependencies.auth
+     * @param {Object} dependencies.apiMail
+     * @param {Object} dependencies.apiSettings
      */
-    constructor({dbBackup, models}) {
+    constructor({dbBackup, models, auth, apiMail, apiSettings}) {
         this.dbBackup = dbBackup;
         this.models = models;
+        this.auth = auth;
+        this.apiMail = apiMail;
+        this.apiSettings = apiSettings;
+    }
+
+    async resetAllPasswords(frameOptions) {
+        return this.models.Base.transaction(async (t) => {
+            frameOptions.transacting = t;
+
+            // Reset all passwords
+            const users = await this.models.User.findAll({...frameOptions});
+            for (const user of users) {
+                await user.save({
+                    status: 'locked' // Prevent signins before password reset
+                }, {...frameOptions});
+            }
+
+            // Delete all sessions
```

Comment:
I don't think this whole block belongs here. Anything to do with session data processing imo should be in the [SessionStore](https://github.com/TryGhost/Ghost/blob/44053553f5baf7a2a1e2f688ceef235afb9e3b82/core/server/services/auth/session/store.js). 

The session store base class we extend in Ghost has exactly the method that is mean to remove all the session data - https://github.com/expressjs/session/tree/0048bcac451ad867299d404aca94c79cc8bc751d#storeclearcallback.

Sidenote, maybe worth looking into truncating the table instead of doing a findAll-> iterate delete dance :thinking:

Reply:
The two reasons I had chosen the findAll->iterate are:
1. It's only touching the staff users, so it shouldn't be more than a couple hundreds sessions
2. This way it triggers hooks if we want to use them in the future.

I'm fine with doing a truncate though, it's optimizing performance over the hooks feature which is ok.

Regarding the code placement, I've moved everything to the express session object but I had to export the express out of the session service, so let me know what you think about it.